db/remote/db.go

```diff
@@ -13,11 +14,11 @@ import (
 var _ db.DB = (*DB)(nil)
 
 type DB struct {
-	ctx context.Context
-
+	ctx        context.Context
 	grpcClient *grpc.ClientConn
 	kvClient   gen.KVClient
 	log        utils.SimpleLogger
+	listener   db.EventListener
```

Comment:
there are some improvements that we can do:
1. set `listener` to `SelectiveListener` in `New` function by default
2. remove nil checks for `d.listener`